extensions/amp-auto-lightbox/0.1/amp-auto-lightbox.js

```diff
@@ -366,14 +373,18 @@ function generateLightboxUid() {
  * @param {!../../../src/service/ampdoc-impl.AmpDoc} ampdoc
  * @param {!Element} element
  * @return {!Promise<!Element>}
+ * @visibleForTesting
  */
-function apply(ampdoc, element) {
+export function apply(ampdoc, element) {
   return Mutation.mutate(element, () => {
     element.setAttribute(LIGHTBOXABLE_ATTR, generateLightboxUid());
   }).then(() => {
     Services.extensionsFor(ampdoc.win)
         .installExtensionForDoc(ampdoc, REQUIRED_EXTENSION);
 
+    element.dispatchEvent(createCustomEvent(ampdoc.win,
```

Comment:
a lot of other components listen to this, can we batch it? just fire once after all of `scan->apply` cycle is done.

Reply:
The scan-apply cycle won't finish until all the scanned images have been loaded, so that won't quite work. Alternatively we can post an event specific to this use case so that `DOM_UPDATE` handlers don't get triggered. 

(Though in practice the event won't be fired many times in one loop, per different images loading at different times and not many instances of lightbox criteria met in the real world.)

Reply:
I looked at `amp-bind` and it does not dispatch this when attributes change. This event is really meant for when something new is added to the DOM (at least assumptions in the code is that way). Let's have a different event for this. We can even tell lightbox-gallery the exact element it needs to make lighboxable so it doesn't even need to scan.

Reply:
Done, PTAL.